adafruit_display_shapes/polygon.py

```diff
@@ -33,121 +33,162 @@
 
 
 class Polygon(displayio.TileGrid):
-    # pylint: disable=too-many-arguments,invalid-name
     """A polygon.
 
     :param list points: A list of (x, y) tuples of the points
     :param int|None outline: The outline of the polygon. Can be a hex value for a color or
                     ``None`` for no outline.
+    :param bool close: (Optional) Wether to connect first and last point. (True)
+    :param int colors: (Optional) Number of colors to use. Most polygons would use two, one for
+                    outline and one for fill. If you're not filling your polygon, set this to 1
+                    for smaller memory footprint. (2)
     """
 
+    _OUTLINE = 1
+    _FILL = 2
+
     def __init__(
         self,
         points: List[Tuple[int, int]],
         *,
         outline: Optional[int] = None,
+        close: Optional[bool] = True,
+        colors: Optional[int] = 2,
     ) -> None:
-        xs = []
-        ys = []
-
-        for point in points:
-            xs.append(point[0])
-            ys.append(point[1])
+        (x_s, y_s) = zip(*points)
 
-        x_offset = min(xs)
-        y_offset = min(ys)
+        x_offset = min(x_s)
+        y_offset = min(y_s)
 
         # Find the largest and smallest X values to figure out width for bitmap
-        width = max(xs) - min(xs) + 1
-        height = max(ys) - min(ys) + 1
+        width = max(x_s) - min(x_s) + 1
+        height = max(y_s) - min(y_s) + 1
 
-        self._palette = displayio.Palette(3)
+        self._palette = displayio.Palette(colors + 1)
         self._palette.make_transparent(0)
-        self._bitmap = displayio.Bitmap(width, height, 3)
+        self._bitmap = displayio.Bitmap(width, height, colors + 1)
+
+        shifted = [(x - x_offset, y - y_offset) for (x, y) in points]
 
         if outline is not None:
-            # print("outline")
             self.outline = outline
-            for index, _ in enumerate(points):
-                point_a = points[index]
-                if index == len(points) - 1:
-                    point_b = points[0]
-                else:
-                    point_b = points[index + 1]
-                self._line(
-                    point_a[0] - x_offset,
-                    point_a[1] - y_offset,
-                    point_b[0] - x_offset,
-                    point_b[1] - y_offset,
-                    1,
-                )
+            self.draw(self._bitmap, shifted, self._OUTLINE, close)
 
         super().__init__(
             self._bitmap, pixel_shader=self._palette, x=x_offset, y=y_offset
         )
 
-    # pylint: disable=invalid-name, too-many-locals, too-many-branches
+    @staticmethod
+    def draw(
```

Comment:
I'm curious about a few of these methods being static. `draw()`, `_line_on()` and `_safe_draw()`  Did this change to static factor in to memory efficiency somehow?

Is there a situation in which they are intended for use outside of in an instance of this class ultimately inside of a sparkline? 

My first inclination was that they could be regular functions and could access bitmap from self rather than having it passed in separately. That way they are always acting upon only the bitmap that is part of this instance.

Reply:
I see resposibilities as follows:
- `Polygon` class: class containing logic of drawing polygons
- `polygon` object: bitmap containing single drawn polygon

Having `draw()` method as static alows us to reuse drawing logic to draw multiple times on the same bitmap - which allows for two things previously not possible:
- having multiple polygons on single bitmap (lower memory usage)
- reusing same bitmap for sequential redrawing of different polygons on the same bitmap (common case for sparklines, and not recreating bitmap over and over potentially means less fragmentation).